detectors/gcp/cloud-run_test.go

```diff
@@ -0,0 +1,164 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gcp
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/google/go-cmp/cmp"
+
+	"go.opentelemetry.io/otel/label"
+	"go.opentelemetry.io/otel/sdk/resource"
+)
+
+var (
+	notOnGCE = func() bool { return false }
+	onGCE    = func() bool { return true }
+)
+
+func getenv(m map[string]string) func(string) string {
+	return func(s string) string {
+		if m == nil {
+			return ""
+		}
+		return m[s]
+	}
+}
+
+type client struct {
+	m map[string]string
+}
+
+func (c *client) Get(s string) (string, error) {
+	got, ok := c.m[s]
+	if !ok {
+		return "", fmt.Errorf("%q do not exist", s)
+	} else if got == "" {
+		return "", fmt.Errorf("%q is empty", s)
+	}
+	return got, nil
+}
+
+func (c *client) InstanceID() (string, error) {
+	return c.Get("instance/id")
+}
+
+func (c *client) ProjectID() (string, error) {
+	return c.Get("project/project-id")
+}
+
+var _ metadataClient = (*client)(nil)
+
+func TestCloudRunDetector_NotOnGCE(t *testing.T) {
```

Comment:
Can we remove the underscores from the test function names?

Reply:
Fixed.